ble/UUID.h

```diff
@@ -91,10 +165,17 @@ class UUID {
     /**
      * Fill in a 128-bit UUID; this is useful when the UUID isn't known at the time of the object construction.
      */
-    void setupLong(const LongUUIDBytes_t longUUID) {
+    void setupLong(const LongUUIDBytes_t longUUID, BitOrder_t order = UUID::MSB) {
         type      = UUID_TYPE_LONG;
-        memcpy(baseUUID, longUUID, LENGTH_OF_LONG_UUID);
-        shortUUID = (uint16_t)((longUUID[2] << 8) | (longUUID[3]));
+        if (order == UUID::MSB) {
+            // Switch endian. Input is big-endian, internal representation is little endian.
+            for (size_t index = 0; index < LENGTH_OF_LONG_UUID; index++) {
```

Comment:
just a side note, the std library provide a wonderful algorithm for this task: `std::reverse_copy`.